src/App.js

```diff
@@ -1,63 +1,103 @@
-// App.js
 import React, { useEffect, useState } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-
 function App() {
-
-  const [todoList, setTodoList] = useState([]);
-  const [isLoading, setIsLoading] = useState(true);
-
-
-
-  useEffect(() => {
-    new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [], // Set todoList property with data from localStorage
-          }
-        });
-      }, 2000); // Delay of 2000 milliseconds (2 seconds)
-    }).then((result) => {
-      setTodoList(result.data.todoList); // Update the todoList state with the fetched data
-      setIsLoading(false); // Turn off loading indicator
-    });
-  }, []); // Empty dependency array, so this effect runs once after initial render
-
-
-    useEffect(() => {
-      if (!isLoading) {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList)); //save the todoList as a string in the localStorage whenever it changes 
+  // Define states for the list of todos and loading state
+  const [todoList, setTodoList] = useState([]); // Initialize todoList state with an empty array
+  const [isLoading, setIsLoading] = useState(true); // Initialize isLoading state as true
+  const [appState, setAppState] = useState ('')
+
+  async function fetchData() {
+    // Define options for the API request
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+      },
+    };
+
+    // Construct the URL for the API
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
+
+    try {
+      // Fetch data from the API using the constructed URL and options
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
       }
-    }, [todoList, isLoading]); // execute/call the callback function(effect) only when the 'todoList' state variable changes by specifying 'todoList' as the dependency array
-
 
+      const data = await response.json();
 
+      // Transform API data.records into todo objects
+      const todos = data.records.map((record) => ({
+        id: record.id,
+        title: record.fields.title,
+      }));
 
-    function removeTodo(id) {
-      // Filter the todoList array to exclude the todo item with the specified id
-      const updatedTodoList = todoList.filter((todo) => todo.id !== id);
+      // Update the todoList state with fetched todos
+      setTodoList(todos);
+      setIsLoading(false); // Set isLoading to false after data fetch
 
-      // Update the todoList state with the new array of todos
-      setTodoList(updatedTodoList);
+    } catch (error) {
+      console.error('Error fetching data:', error.message);
     }
```

Comment:
This function no longer existing is one of the things that's causing a page crash